addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -38,6 +38,16 @@ type ManagedClusterAddOnSpec struct {
 	// +kubebuilder:validation:MaxLength=63
 	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
 	InstallNamespace string `json:"installNamespace,omitempty"`
+
+	// disableLeaseCheck indicates whether the registration agent should check the lease of the
+	// addon on the managed cluster.
+	// By default, the registration agent maintains the addon available status by checking the
+	// lease of the addon on the managed cluster, so the addon agent should maintain a lease in
+	// its installation namespace with its status.
+	// If this field sets to true, the registration agent will not check the lease of the addon on
+	// the managed cluster, the addon should maintain its available status by itself
+	// +optional
+	DisableLeaseCheck bool `json:"disableLeaseCheck,omitempty"`
```

Comment:
https://github.com/open-cluster-management-io/addon-framework/pull/51

as an alternative of lease-based healthcheck, i think we can have another type of checker which is mapping the status of corresponding work resource to the addon, which can also be done in the registration controller. so it's probably better to generalize a classful healthcheck e.g. including `None`, `Lease`, `Work` b/c it's more extensible. btw i am also sort of worrying if it's a good idea to make the registration controller a centralized healthchecker in the long term, considering that it can be a potential scalability bottleneck. besides, imagine a case where a singleton healthcheck is temporarily down in maintainence mode, will all the addons be down after that?

Reply:
+1, a healthcheck type looks more extensible and more clear than a bool flag

Reply:
I would want this as a struct with a mode in it.

Reply:
introduce a struct to represent this